apps/web/app/(app)/environments/[environmentId]/settings/tags/actions.ts

```diff
@@ -1,38 +1,47 @@
 "use server";
 
 import { deleteTag, mergeTags, updateTagName } from "@formbricks/lib/tag/service";
-import { canUserAccessTag } from "@formbricks/lib/tag/auth";
+import { canUserAccessTag, verifyUserRoleAccess } from "@formbricks/lib/tag/auth";
 import { authOptions } from "@formbricks/lib/authOptions";
 import { getServerSession } from "next-auth";
 import { AuthorizationError } from "@formbricks/types/errors";
 
-export const deleteTagAction = async (tagId: string) => {
+export const deleteTagAction = async (tagId: string, environmentId: string) => {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessTag(session.user.id, tagId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasDeleteAccess } = await verifyUserRoleAccess(environmentId, session.user.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.